packages/mongo/collection.js

```diff
@@ -451,8 +451,11 @@ Mongo.Collection = class Collection {
       throw new Error("insert requires an argument");
     }
 
-    // Shallow-copy the document and possibly generate an ID
-    doc = { ...doc };
+    // Make a shallow clone of the document, preserving its prototype.
+    doc = Object.create(
+      Object.getPrototypeOf(doc),
+      Object.getOwnPropertyDescriptors(doc)
+    );
```

Comment:
This idiom deserves a signal boost. I could've squashed this commit into the previous one, but I didn't want this bug fix to be buried, because the underlying problem is worth understanding.